src/SeqCli/Forwarder/Storage/BufferAppenderChunk.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright Datalust Pty Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using SeqCli.Forwarder.Filesystem;
+
+namespace SeqCli.Forwarder.Storage;
+
+sealed record BufferAppenderChunk(StoreFileAppender Appender) : IDisposable
```

Comment:
Records are normally value-like; it'd be strange for the generated equality/hash implementation etc. on this type to include `Appender`. Should probably just be `class`?